### PR TITLE
Add glossary of terms to handbook

### PR DESCRIPTION
**Pull request by jasonyavorsky**
From thread at https://sourcegraph.slack.com/archives/C89KCDK5J/p1644262327977789. Source is https://docs.google.com/spreadsheets/d/1E71lKEMl6sFvey0BAVWyQih2an64bSutZfBJ6haXY9c/edit#gid=0

Implements the terms as YAML so they can be used by the `sg` tool.

### CODE REVIEW

**Comment by jasonyavorsky** [comment]
@InesRoitman once this is merged it's worth adding to onboarding materials.

**Comment by InesRoitman** [comment]
> @InesRoitman once this is merged it's worth adding to onboarding materials.

@jyavorska Definitely, thanks! We currently have [this glossary](https://handbook.sourcegraph.com/company-info-and-process/onboarding/glossary) in the onboarding process.

**Comment by jasonyavorsky** [comment]
Oh that's great @InesRoitman I combined this with that one so there aren't two separate ones.

File: content/company-info-and-process/onboarding/glossary.md
Comment (jasonyavorsky) [inline content/company-info-and-process/onboarding/glossary.md:7]:
```suggestion
# Developer lingo for non-developers: a guide
```

**Comment by bobheadxi** [comment]
@jyavorska FYI there's some discussion in another PR about maybe making this a handbook `data` file e.g. `glossary.yaml`, and some debate about whether we should have this at all:

- https://github.com/sourcegraph/sourcegraph/pull/30785#discussion_r801171176
- https://github.com/sourcegraph/sourcegraph/pull/30785#discussion_r801180935

**Comment by jasonyavorsky** [comment]
Thanks for sharing @bobheadxi. To me the handbook feels more accessible to more people, but I'm fine with whatever everyone decides.

**Comment by courier-new** [comment]
Hey @jyavorska!! I love the idea of including this in the handbook, and personally I could really get behind @bobheadxi's [suggestion](https://github.com/sourcegraph/sourcegraph/pull/30785#discussion_r801171176) of having it live in the handbook as a YAML file that powers both a page in the handbook as well as an sg command. I'm pretty low on bandwidth for the rest of this week, so do you think you could help me out with facilitating that? 🙂 Could we use this PR to take the markdown dictionary, convert it to a YAML file, and generate a handbook page from it? Then the sg command could source definitions from the YAML file instead.

**Comment by jasonyavorsky** [comment]
Sure, done @courier-new

**Comment by jasonyavorsky** [comment]
@bobheadxi @slimsag @courier-new who would decide if this is mergeable or not?

**Review by bobheadxi — APPROVED** [review]
LGTM, this is great 😁 Might want to let @courier-new give an approval as well since I think she's working on the `sg` side of this!

**Review by courier-new — APPROVED** [review]
I'll give this my approval, anyway. There was some discussion on my PR around if formalizing this actually creates an incentive to use acronyms and jargon *more often*, or at least pushes the needle further in that direction. I personally think some amount of acronyms/jargon is never going away and can even be helpful, though. And a lot of these aren't Sourcegraph-specific terms, so even if we avoid using them here, this still is a useful resource for encountering them "in the wild." I think we can do more to discourage their use in general, but providing this glossary empowers teammates to figure out what things mean on their own, so I'm strongly in favor.

**Comment by bobheadxi** [comment]
> I personally think some amount of acronyms/jargon is never going away and can even be helpful [...] we can do more to discourage their use in general, but providing this glossary empowers teammates to figure out what things mean on their own, so I'm strongly in favor.

Big +1 👍

**Comment by jasonyavorsky** [comment]
Ok great, merging! Thanks everyone. Your feedback definitely made this better in the end.

I did add some more text at one point about avoiding acronyms.

**Pull request merged**
